File: client/api/megaphone.ts
import { info } from "@trippler/tr_lib/shared"
import { toggleMegaphone } from "../modules/megaphone"

export const Bridge = {
  drawtext: {
    show: (key: string | null, text: string) => {
      info`you haven't set your drawtext function yet :(`
    },
    hide: () => {
      info`you haven't set your drawtext function yet :(`
    },
  },
  voice: {
    clearProximityOverride: exports["pma-voice"].clearProximityOverride(),
    overrideProximityRange: (range: number, _: boolean) => exports["pma-voice"].overrideProximityRange(range, _)

  },
  notify: (message: string, type?: string, duration?: number) => exports['qb-core'].Notify(message, type, duration)
}

onNet('lenix_patrolmegaphone:client:toggle', toggleMegaphone)

File: lenixm/lenix_patrol/client/api/megaphone.lua
local megaphoneMod<const> = require 'client/modules/megaphone'

local Bridge<const> = {
  drawtext = {
    show = function(key, text)
      lib.info('you haven\'t set your drawtext function yet :(')
    end,
    hide = function()
      lib.info('you haven\'t set your drawtext function yet :(')
    end,
  },
  voice = {
    clearProximityOverride = function()
      exports["pma-voice"]:clearProximityOverride()
    end,
    overrideProximityRange = function(range, _)
      exports["pma-voice"]:overrideProximityRange(range, _)
    end
  },
  notify = function(message, type, duration)
    exports['qb-core']:Notify(message, type, duration)
  end
}

RegisterNetEvent('lenix_patrolmegaphone:client:toggle', function()
  megaphoneMod.toggleMegaphone()
end)

return Bridge